src/core/or/connection_or.c

```diff
@@ -705,6 +705,7 @@ connection_or_finished_connecting(or_connection_t *or_conn)
   log_debug(LD_HANDSHAKE,"OR connect() to router at %s:%u finished.",
             conn->address,conn->port);
   control_event_bootstrap(BOOTSTRAP_STATUS_HANDSHAKE, 0);
+  control_event_boot_first_orconn();
```

Comment:
So the bootstrap process starts with **connecting** to a directory server (5%). Then, right here at this point, the boostrap will change to `Finishing handshake with directory server` (10%). With this addition, we'll flag that we have a "first connect() to a OR".

Which means that we can't go pass 10% in the first place if we haven't at least connected to our directory server right? Which makes me wonder if this will really delay any directory information bootstrap progress if our first connect to an OR conn is always at the 10% mark? Won't we be stuck at 5% if we are unable to connect to an  OR? (in this case at 5%, the directory server).

Reply:
I think being stuck at 5% or 10% gives the user a clearer indication of what's wrong than being stuck at 80%.

>Which means that we can't go pass 10% in the first place if we haven't at least connected to our directory server right?

Empirically, if there is enough cached directory info, bootstrap will jump from 0% to 80% without necessarily being able to connect to _any_ relay or bridge. I can paste example logs of "warm cache" cases both before and after these patches, if that helps. (probably to the Trac ticket though)

Reply:
> Empirically, if there is enough cached directory info, bootstrap will jump from 0% to 80% without necessarily being able to connect to _any_ relay or bridge. I can paste example logs of "warm cache" cases both before and after these patches, if that helps. (probably to the Trac ticket though)

Oh ok I see the jump from 0 to 80 but stuck at the connecting to OR step which is not fun for the user.